### ISSUE TITLE
sql-sync options doens't seem to be picked up from some.site.yml

### ISSUE THREAD

**Issue by hanoii** [drush-ops/drush#3421]
This doesn't work

```!yml
develop:
  command:
    sql:
      sync:
        options:
          source-dump: /tmp/dump
```
But this, does:

`drush sql-sync @site.develop @self --source-dump=/tmp/dump`

`Drush version : 9.2.1`

**Comment by greg-1-anderson** [comment]
Did you try:
```
command:
  sql:
    sync:
      options:
        source-dump: /tmp/dump
```

**Comment by hanoii** [comment]
@greg-1-anderson not sure I understand what you meant. I tried exactly that as I put in the first comment, or did you mean something else?

**Comment by greg-1-anderson** [comment]
In your comment, `develop:` seems out of place. Try without that. I did not try to reproduce myself.

**Comment by greg-1-anderson** [comment]
Oh, this is an alias. I was thinking this was a drush.yml file. Need to consider again.

**Comment by greg-1-anderson** [comment]
So, `drush @site.env command` will inject the options from @site.env for command. I'll have to look and see what sql-sync does with alias arguments; it may be less limited.

**Comment by hanoii** [comment]
sure.. yes it's an alias...  it's odd though, at least based on what what `example.site.yml` says:

```yml
# - 'command': These options will only be set if the alias
#   is used with the specified command.  In the example below, the option
#   `--no-dump` will be selected whenever the @stage alias
#   is used in any of the following ways:
#     - `drush @stage sql-sync @self @live`
#     - `drush sql-sync @stage @live`
#     - `drush sql-sync @live @stage`
```

```
# stage:
#   uri: http://stage.example.com
#   root: /path/to/remote/drupal/root
#   host: mystagingserver.myisp.com
#   user: publisher
#   os: Linux
#   paths:
#    - files: sites/mydrupalsite.com/files
#    - custom: /my/custom/path
#   command:
#     sql:
#       sync:
#         options:
#           no-dump: true
```

it implies that what I am doing should work, it's not.

Awesome if you can look.

**Comment by greg-1-anderson** [comment]
Looking at the code, it appears that the documentation you quoted above was converted from Drush 8 (which does support this feature), but the code to implement it was never implemented in Drush 9.

I don't think it will be hard to fix.

**Comment by weitzman** [comment]
Also that doc should not use a boolean value since those dont work.

**Comment by hanoii** [comment]
@weitzman I didn't copied it completely, it follows:

```yml
#   NOTE: Setting boolean options broke with Symfony 3. This will be fixed
#     in a future release. See: https://github.com/drush-ops/drush/issues/2956
```

**Comment by jurgenhaas** [comment]
Came across a similar issue, but with the drush.yml configuration file. The documentation states it should be coded like this:

```
command:
  sql:
    dump:
      options:
        sync:
          options:
```

which seems to be wrong. The correct version would be

```
command:
  sql:
    sync:
      options:
```

However, when I use the option `structure-tables-key: 'common'` this will be used correctly where the option `skip-tables-key: 'common'` is not being used at all. When I use `--skip-tables-key=common` as a CLI option, then it is working as expected.

Is this related here or do I have to open a new issue?

**Comment by greg-1-anderson** [comment]
Hm, there isn't any reason I can think of why `structure-tables-key` should behave differently than `skip-tables-key`, but I haven't looked at the code to confirm. Could have something to do with how sql-sync calls sql-dump.

**Comment by greg-1-anderson** [comment]
Or maybe sql-sync has not named all of the sql-dump options in its option list.

**Comment by jurgenhaas** [comment]
If I remember correctly, this had already been a similar issue in Drush 8 and I just had been lazy to report it at the time and resolved it in my shell alias. But since those are no longer supported I'm now stuck with that issue and now wonder if it may be a left over issue from earlier versions?

**Comment by weitzman** [comment]
There may be a bug here, but I can't think of a reason to store an option like structure-tables-key in an alias. Its not likely to vary by environment. Therefore, put it in a config file like /drush/drush.yml. At least thats what I do on my big Drush9 project.

**Comment by jurgenhaas** [comment]
@weitzman yes, but: it doesnt work in the drush.yml file right now as I reported late in my comments. It's consistant and reproducable. Do you need more detail?

**Comment by weitzman** [comment]
I did some testing, and am unable to reproduce this. Both --structure-tables-key and --skip-tables-key are being used by sql:dump. Note that these keys have to defined on sql:dump, not sql:sync. This is clarified in a new docs commit 7fbf122. Thanks for pointing out that this was malformed.

Let us know if the new docs help get this working.

**Comment by hanoii** [comment]
I should point out that this issue somehow diverged in its topic. My initial issue is still there, which relates to being able to pass options to sql-sync through the alias file.

**Comment by weitzman** [comment]
@hanoii - are you using 9.2.1 on both sides of the sql:sync? if so you shouldn't need to define a dump file. the name is randomly generated for you. 

the issue here is that source-dump is looked up on the machine which is initiating the the sql:sync which is @self, not \@develop. this may be different than prior versions of drush, which is what @greg-1-anderson is alluding to in https://github.com/drush-ops/drush/issues/3421#issuecomment-369382483.

We need to fix docs or add back this feature. I'm ambivalent.

**Comment by greg-1-anderson** [comment]
We need a site-alias API to adjust `$options` based on the provided alias parameters.

**Comment by weitzman** [comment]
If HOME isn't writable on a host, you might as well point drush to a writable home. If possible, set `HOME=/tmp`. Thats an environment variable called HOME ([see usage](https://github.com/drush-ops/drush/blob/fac6079d17201f65e1b6d1b9f38862e33394889e/drush.php#L57)). 

One technique for setting env variables [is now documented in a PR at drupal-project](https://github.com/drupal-composer/drupal-project/pull/351). You can implement that PR on your own project, even before the PR is merged.
